pkg/kubectl/cmd/resource/get.go

```diff
@@ -221,6 +235,51 @@ func (options *GetOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args
 			return cmdutil.UsageErrorf(cmd, usageString)
 		}
 	}
+
+	// TODO(juanvallejo): this needs cleanup. Fields for a nested PrintFlags struct should not be set here
+	options.HumanPrintFlags.WithNamespace = options.AllNamespaces
+	options.HumanPrintFlags.AbsoluteTimestamps = options.Watch
+
+	printer, matches, err := options.HumanPrintFlags.ToPrinter(options.OutputFormat)
+	if !matches {
+		return fmt.Errorf("unable to match a printer to handle current print options")
```

Comment:
@soltysh I am still not sure how to handle this error better / if we should fallback instead?

Reply:
For now (at least for this prototype) I fallback to using `cmdutil.PrinterForOptions(cmdutil.ExtractPrintOptions(cmd, ...))`.

Once we wire printers at the command-level, that might no longer be an option.

If no printer were to match, then I'd consider that a programmer error, but not sure how to best present that through an error in this spot.